### ISSUE TITLE
Pass "requestId" as a response resolver argument

### ISSUE THREAD

**Issue by skvale** [kettanaito/msw#1933]
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

In MSW v1 I was able to access `req.id` in a handler function which corresponded with the requestId in events.

For example
```ts
worker.events.on('response:mocked', ({requestId}) => {
// and
rest.post('my-url', (req) => { req.id 
```

were the same

In v2, the request object doesn't have an id attribute.  From what I can tell the requestId is unique to events

This seems like a nice to have addition to the handler args that wouldn't be hard to add

Here's what I had in mind as a diff - https://github.com/mswjs/msw/compare/main...skvale:msw:main

**Comment by kettanaito** [comment]
Hi, @skvale. Thanks for suggesting this. 

It slipped from my attention that we've removed `req.id` without a substitute. I think what you are proposing makes sense. My only comment here would be to expose the variable as `requestId`, both to keep the context and stay consistent with the same variable in the Life-cycle event listener arguments. 

Would you be interested in opening a pull request for this? I can help you with the tests.

**Comment by skvale** [comment]
PR: https://github.com/mswjs/msw/pull/1942

**Comment by kettanaito** [comment]
## Released: v2.1.0 🎉

This has been released in v2.1.0!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.1.0)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.1.0)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
